pydarn/plotting/rtp.py

```diff
@@ -26,22 +29,602 @@ class RTP(matplotlib.pyplot):
+        # TODO: implement variant other coordinate systems for the y-axis
+        y = np.linspace(0, cls.dmap_data[0]['nrang'],
+                        cls.dmap_data[0]['nrang'])
+        y_max = cls.dmap_data[0]['nrang']
+
+        # z: parameter data mapped into the color mesh
+        z = np.zeros((1, y_max)) * np.nan
+
+        # x: time date data
+        x = []
+
+        # We cannot simply use numpy's built in min and max function
+        # because of the ground scatter value
+        if settings["boundary"]:
+            z_min, z_max = settings["boundary"]
+        else:
+            z_min = cls.dmap_data[0][parameter][0]
+            z_max = cls.dmap_data[0][parameter][0]
+        for dmap_record in cls.dmap_data:
+            # get time difference to test if there is some gap data
+            time = cls.__time2datetime(dmap_record)
+            diff_time = 0.0
+            if time > end_time:
+                break
+            if x != []:
+                # 60.0 seconds in a minute
+                delta_diff_time = (time - x[-1])
+                diff_time = delta_diff_time.seconds/60.0
+
+            # separation roughly 2 minutes
+            if diff_time > 2.0:
+                # if there is gap data (no data recorded past 2 minutes)
+                # then fill it in with white space
+                for _ in range(0, int(np.floor(diff_time/2.0))):
+                    x.append(x[-1] + timedelta(0, 120))
+                    i = len(x) - 1  # offset since we start at 0 not 1
+                    if i > 0:
+                        z = np.insert(z, len(z), np.zeros(1, y_max) * np.nan,
+                                      axis=0)
+
+            # Get data for the provided beam number
+            if (beam_num == 'all' or dmap_record['bmnum'] == beam_num) and\
+               (settings['channel'] == 'all' or
+                    dmap_record['channel'] == settings['channel']):
+                if start_time <= time:
+                    # construct the x-axis array
+                    # Numpy datetime is used because it properly formats on the
+                    # x-axis
+                    x.append(time)
+                    # I do this to avoid having an extra loop to just count how
+                    # many records contain the beam number
+                    i = len(x) - 1  # offset since we start at 0 not 1
+
+                    # insert a new column into the z_data
+                    if i > 0:
+                        z = np.insert(z, len(z), np.zeros(1, y_max) * np.nan,
+                                      axis=0)
+                    try:
+                        # get the range gates that have "good" data in it
+                        for j in range(len(dmap_record['slist'])):
+                            # if it is ground scatter store a very
+                            # low number in that cell
+                            if settings['ground_scatter'] and\
```

Comment:
typically groundscatter is one word, would probably be best to get rid of the underscore.

Reply:
done